src/Microsoft.Azure.WebJobs.Host/Singleton/SingletonListener.cs

```diff
@@ -101,6 +98,10 @@ public void Dispose()
                 LockTimer.Dispose();
             }
 
+            // When we Dispose, it's important that we release the lock if we
+            // have it.
+            ReleaseLockAsync().Wait();
```

Comment:
In general, any call to .Wait() in can deadlock. This is a particularly common issue in ASP.Net due to the sync context.  :|  
The GetAwaiter() goo compensates for it. 
A good explanation is at: [http://blog.stephencleary.com/2012/07/dont-block-on-async-code.html](http://blog.stephencleary.com/2012/07/dont-block-on-async-code.html)

---
In reply to: [63921309](https://github.com/Azure/azure-webjobs-sdk/pull/728#discussion_r63921309) [](ancestors = 63921309)